src/main/java/apoc/metrics/Metrics.java

```diff
@@ -125,6 +179,50 @@ public Neo4jMetric(String name, long lastUpdated) {
         return new LoadCsv().csv(url, config);
     }
 
+    @Procedure(mode=Mode.DBMS)
+    @Description("apoc.metrics.storage(directorySetting) - retrieve storage metrics about the devices Neo4j uses for data storage. " +
+            "directorySetting may be any valid neo4j directory setting name, such as 'dbms.directories.data'.  If null is provided " +
+            "as a directorySetting, you will get back all available directory settings.  For a list of available directory settings, " +
+            "see the Neo4j operations manual reference on configuration settings.   Directory settings are **not** paths, they are " +
+            "a neo4j.conf setting key name")
+    public Stream<StorageMetric> storage(@Name("directorySetting") String directorySetting) {
+        // Permit case-insensitive checks.
+        String input = directorySetting == null ? null : directorySetting.toLowerCase();
+
+        boolean validSetting = (
+            input == null ||
+            Arrays.stream(neo4jDirectoryConfigurationSettingNames).anyMatch(input::equals)
```

Comment:
contains?

Reply:
Don't think so...no.   If the input contains the valid setting, this means you could ask for storage metrics for "dbms.directories.logs@)#&*)&!)#$&!)#&$!#" --- not sure that makes sense, isn't it better to give an error in this case.

Reply:
contains as in list.contains(setting) not string contains :)